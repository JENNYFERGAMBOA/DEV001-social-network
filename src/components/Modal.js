export const closeClick = (modal, closeModalId, e) => {
  if ((e.target === modal) || (e.target.getAttribute('id') === closeModalId)) {
    modal.style.display = 'none';
  }
};

export const Modal = (title, message) => {
  const $modalDiv = document.createElement('div');
  $modalDiv.id = 'myModal';
  $modalDiv.className = 'modal';
  const $closeId = 'modal-close';
  $modalDiv.innerHTML = `
  <div class="modal__content"> 
  <span class="modal__close" id="${$closeId}">&times;</span>
  <h2>${title}</h2> 
  <p>${message}</p> </div>`;

  document.body.appendChild($modalDiv);
  $modalDiv.style.display = 'block';
  const closeModal = document.getElementById($closeId);

  // When the user clicks on <span> (x), close the modal
  window.addEventListener('click', (e) => {
    closeClick($modalDiv, closeModal.id, e);
  });

  return $modalDiv;
};
