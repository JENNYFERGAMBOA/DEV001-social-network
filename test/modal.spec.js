import {Modal } from '../src/components/Modal.js';
/* import { modalOpen } from '../src/controllers/c-modal'; */

// const element = Modal();

describe('test function Modal', () => {
  /* const contentModal = element.querySelector('.modal__content'); */
  it('Should test if the modal element exists', () => {
    expect(Modal('hola soy modal', 'hola soy el mensaje de modal')).toMatchSnapshot();
  });

  it('Should check that the modal closes', () => {
    const element = Modal('hola soy modal', 'hola soy el mensaje de modal');
    const closeModal = document.getElementById('modal-close');
    const e = new Event('click');
    closeModal.dispatchEvent(e);
    // closeClick(element, closeModal.id);
    // closeClick(element, closeModal, e.target);
    expect(element.style.display).toBe('none');
  });
});
