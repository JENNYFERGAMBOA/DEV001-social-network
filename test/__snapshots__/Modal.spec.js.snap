// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test function Modal Probar si existe el elemento modal 1`] = `
<div
  class="modal"
  id="myModal"
>
  
 
  <div
    class="modal__content"
  >
     
  
    <span
      class="modal__close"
      id="modal-close"
    >
      Ã—
    </span>
    
  
    <h2>
      hola soy modal
    </h2>
     
  
    <p>
      hola soy el mensaje de modal
    </p>
     
  </div>
</div>
`;
